applications/crossbar/src/modules/cb_modules_util.erl

```diff
@@ -288,7 +289,7 @@ originate_quickcall(Endpoints, Call, Context) ->
     Request = [{<<"Application-Name">>, <<"transfer">>}
                ,{<<"Application-Data">>, get_application_data(Context)}
                ,{<<"Msg-ID">>, MsgId}
-               ,{<<"Endpoints">>, maybe_auto_answer(Endpoints, AutoAnswer)}
+               ,{<<"Endpoints">>, set_outbound_call_id(CallId, maybe_auto_answer(Endpoints, AutoAnswer))}
```

Comment:
Maybe you could refactor this to not iterate over the list of endpoints twice? Convert it to a fold and run the `set_value` calls as functions to the fold?